### PR TITLE
Avoid using Identifiable and CloneablePtr

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli or @geirst PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@vekterli Easter did not make it go away....

File: document/src/vespa/document/update/fieldpathupdate.h [hunk 1]
```diff
@@ -17,29 +17,22 @@ class DataType;
...
         Add    = IDENTIFIABLE_CLASSID(AddFieldPathUpdate),
         Assign = IDENTIFIABLE_CLASSID(AssignFieldPathUpdate),
         Remove = IDENTIFIABLE_CLASSID(RemoveFieldPathUpdate)
     };
 
-    void applyTo(Document& doc) const;
+    virtual ~FieldPathUpdate();
 
-    virtual FieldPathUpdate* clone() const = 0;
+    FieldPathUpdateType type() const { return _type; }
```
Comment (vekterli) [inline document/src/vespa/document/update/fieldpathupdate.h:34]: Consider adding `noexcept`

**Review by vekterli — APPROVED** [review]
:+1:

File: document/src/vespa/document/update/fieldpathupdate.h (hunk 1 above)
Comment (baldersheim) [reply document/src/vespa/document/update/fieldpathupdate.h]: Done

**Pull request merged**